internal/rest/resources/resources.go

```diff
@@ -92,49 +83,66 @@ func ValidateEndpoints(extensionServers map[string]rest.Server, coreAddress stri
 			serverAddresses[server.Address.String()] = true
 		}
 
-		// Ensure no endpoint path conflicts with another endpoint on the same server.
-		// If a server lacks an address, we need to compare it to every other server
-		// that also lacks an address, as well as the internal endpoints.
-		serverPaths, err := resourcesConflict(server, existingEndpointPaths)
+		existingPaths := extensionPaths[serverName]
+		if server.CoreAPI {
+			existingPaths = corePaths
+		}
+
+		err := resourcesConflict(server, existingPaths)
 		if err != nil {
 			return err
 		}
+	}
+
+	return nil
+}
+
+// getServerPaths returns a map of all paths for core and extension servers.
+// the extension server paths are grouped by server name.
+func getServerPaths(extensionServers map[string]rest.Server) (corePaths map[string]bool, extensionPaths map[string]map[string]bool) {
+	baseCoreEndpoints := []rest.Resources{UnixEndpoints, PublicEndpoints, InternalEndpoints}
+	corePaths = make(map[string]bool)
+	extensionPaths = make(map[string]map[string]bool)
+
+	for _, baseEndpoints := range baseCoreEndpoints {
+		for _, e := range baseEndpoints.Endpoints {
+			url := filepath.Join(string(baseEndpoints.PathPrefix), e.Path)
+			corePaths[url] = true
+		}
+	}
+
+	for name, server := range extensionServers {
+		for _, resource := range server.Resources {
+			for _, endpoint := range resource.Endpoints {
+				url := filepath.Join(string(resource.PathPrefix), endpoint.Path)
+				if server.CoreAPI {
+					corePaths[url] = true
+					continue
+				}
 
-		// Update the existing paths now that we have checked a new Server.
-		for k, v := range serverPaths {
-			if !existingEndpointPaths[k] {
-				existingEndpointPaths[k] = v
+				if extensionPaths[name] == nil {
+					extensionPaths[name] = make(map[string]bool)
+				}
+
+				extensionPaths[name][url] = true
```

Comment:
Same here, `extensionPaths[name][url]` might already be `true` which is what we are looking for, but that won't be captured if we overwrite the `true` with another `true`.